### PR TITLE
Release 2.0.0

### PR DESCRIPTION
**Pull request by Smartappli**
Postgresql and Postgis upgrade in version 13 ans 3.1.2 respectivelly.
ugraded scripts:
- New upgrade script (upgrade.sh)
- New options in run.sh (upgrade and planete) to upgrade previous installation and import OSM planet from a dump respectivelly.

Luajit added and compiled with osm2pgsql to speed up import
New index to improve rendering speed

### CODE REVIEW

**Comment by Overv** [comment]
@Istador Do you have ideas for handling PostgreSQL updates like this going forward?

Right now I would simply merge and release something like this as a new major version and call it a day, but perhaps we can provide some kind of automatic upgrade process/instructions.

**Comment by Istador** [comment]
Yeah, it's definitely a breaking change, so it'd become v2.0.0.

For the future, instead of mounting to `/var/lib/postgresql/$version/main` we could introduce a versionless path. E.g. `/data/posgresql` to which we point inside the image from `/var/lib/postgresql/$version/main` via a symlink?

The problem then becomes detecting a version change (afaik this info is somewhere in that folder) and upgrading the database automatically with `pg_upgrade`.

**Comment by Smartappli** [comment]
> 
> 
> Yeah, it's definitely a breaking change, so it'd become v2.0.0.
> 
> For the future, instead of mounting to `/var/lib/postgresql/$version/main` we could introduce a versionless path. E.g. `/data/posgresql` to which we point inside the image from `/var/lib/postgresql/$version/main` via a symlink?
> 
> The problem then becomes detecting a version change (afaik this info is somewhere in that folder) and upgrading the database automatically with `pg_upgrade`.

we could add "upgrade" in run.sh to run pg upgrade.

Don't forget that there is also an upgrade procedure for major versions of postgis

**Comment by Smartappli** [comment]
@Istador & @Overv i have done a propisition of migration script (upgrade.sh) and other improvements.

File: Dockerfile [hunk 1]
```diff
@@ -127,10 +129,11 @@ RUN mkdir -p /home/renderer/src \
 # Configure stylesheet
 RUN mkdir -p /home/renderer/src \
  && cd /home/renderer/src \
- && git clone --single-branch --branch v5.3.1 https://github.com/gravitystorm/openstreetmap-carto.git --depth 1 \
+ && git clone -b master https://github.com/gravitystorm/openstreetmap-carto.git --depth 1 \
```
Comment (Istador) [inline Dockerfile:132]: Please check out a specific version branch and not the `master` if possible. The `master` branch might introduce breaking changes in the future that requires other changes here and in other files.

**Comment by Istador** [comment]
> @Istador & @Overv i have done a propisition of migration script (upgrade.sh) and other improvements.

The problem with the `pg_dump` and import approach is that it doubles the disk space required while executing and doesn't clean up afterwards (remove the `dump.sql`).

Disk space is an issue in this case with large imports of `planet.osm`, that might take up 1.4 TB or more for the database.

---

I'd rather advise to use the `postgres` image with specific versions, instead of assuming that `overv/openstreetmap-tile-server:latest` still is pointing to a postgres 12 version on the host machine:
```
docker run --rm  \
  -v /${CWD}://tmp  \
  -v openstreetmap-data://var/lib/postgresql/12/main  \
  -w //tmp  \
  postgres:12  \
  pg_dumpall -U prostgres > dump.sql

docker run --rm  \
  -v /${CWD}://tmp  \
  -v openstreetmap-data://var/lib/postgresql/13/main  \
  -w //tmp  \
  postgres:13  \
  psql -U postgres -d gis < dump.sql

rm ./dump.sql
```

**Comment by Smartappli** [comment]
> 
> 
> > @Istador & @Overv i have done a propisition of migration script (upgrade.sh) and other improvements.
> 
> The problem with the `pg_dump` and import approach is that it doubles the disk space required while executing and doesn't clean up afterwards (remove the `dump.sql`).
> 
> Disk space is an issue in this case with large imports of `planet.osm`, that might take up 1.4 TB or more for the database.
> 
> I'd rather advise to use the `postgres` image with specific versions, instead of assuming that `overv/openstreetmap-tile-server:latest` still is pointing to a postgres 12 version on the host machine:
> 
> ```
> docker run --rm  \
>   -v /${CWD}://tmp  \
>   -v openstreetmap-data://var/lib/postgresql/12/main  \
>   -w //tmp  \
>   postgres:12  \
>   pg_dumpall -U prostgres > dump.sql
> 
> docker run --rm  \
>   -v /${CWD}://tmp  \
>   -v openstreetmap-data://var/lib/postgresql/13/main  \
>   -w //tmp  \
>   postgres:13  \
>   psql -U postgres -d gis < dump.sql
> 
> rm ./dump.sql
> ```

Thank you for your wise insight. Maybe use the postgis image with postgresql and postgis it might be easier for updates 

docker run --rm  \
  -v /${CWD}://tmp  \
  -v openstreetmap-data://var/lib/postgresql/12/main  \
  -w //tmp  \
  postgis/postgis:12-3.1 \
  pg_dumpall -c -v -U prostgres > dump.sql

docker run --rm  \
  -v /${CWD}://tmp  \
  -v openstreetmap-data://var/lib/postgresql/13/main  \
  -w //tmp  \
  postgis/postgis:13-3.1  \
  psql -U postgres -d gis < dump.sql
  
sudo -u postgres psql -d gis -c "SELECT PostGIS_Extensions_Upgrade();"

rm ./dump.sql

File: Dockerfile [hunk 2]
```diff
@@ -127,10 +129,11 @@ RUN mkdir -p /home/renderer/src \
 # Configure stylesheet
 RUN mkdir -p /home/renderer/src \
  && cd /home/renderer/src \
- && git clone --single-branch --branch v5.3.1 https://github.com/gravitystorm/openstreetmap-carto.git --depth 1 \
+ &&git clone --single-branch --branch v5.3.1 https://github.com/gravitystorm/openstreetmap-carto.git --depth 1 \ git clone --single-branch --branch v5.3.1 https://github.com/gravitystorm/openstreetmap-carto.git --depth 1 \
```
Comment (Istador) [inline Dockerfile:132]: the clone command is duplicated now

File: upgade.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  pg_dumpall -c -v -U prostgres > dump.sql
+
+docker run --rm  \
+  -v /${CWD}://tmp  \
+  -v openstreetmap-data://var/lib/postgresql/13/main  \
+  -w //tmp  \
+  postgres:13  \
+  psql -U postgres -d gis < dump.sql
+  
+sudo -u postgres psql -d gis -c "SELECT PostGIS_Extensions_Upgrade();"
```
Comment (Istador) [inline upgade.sh:19]: this needs to be executed in a docker container with the right database and not on the host (which might not even have psql).

Comment (Smartappli) [reply upgade.sh]: PostGIS upgrade transfered in run.sh

File: Dockerfile (hunk 2 above)
Comment (Smartappli) [reply Dockerfile]: Duplicated code fixed

File: Dockerfile (hunk 1 above)
Comment (Smartappli) [reply Dockerfile]: fixed

**Review by Smartappli — COMMENTED** [review]
all asked modifications done

**Comment by Overv** [comment]
Could you extend the README with instructions of how to perform the upgrade process?

**Comment by Smartappli** [comment]
> 
> 
> Could you extend the README with instructions of how to perform the upgrade process?

ok in 15 minutes

**Comment by Smartappli** [comment]
Add varnish + bortli compression
Add LuaJIT to speed up inportation of 30%

**Pull request closed without merging**